ApplicationCode/ProjectDataModel/RimWellPathLateralGeometryDef.cpp

```diff
@@ -0,0 +1,550 @@
+
+#include "RiaFieldHandleTools.h"
+#include "RiaJCurveCalculator.h"
+#include "RiaLogging.h"
+#include "RiaOffshoreSphericalCoords.h"
+#include "RiaPolyArcLineSampler.h"
+#include "RiaSCurveCalculator.h"
+
+#include "RigWellPath.h"
+
+#include "RimModeledWellPath.h"
+#include "RimProject.h"
+#include "RimWellPathGroup.h"
+#include "RimWellPathTarget.h"
+
+#include "RiuViewerCommands.h"
+
+#include "cafCmdFeatureMenuBuilder.h"
+#include "cafPdmFieldScriptingCapabilityCvfVec3d.h"
+#include "cafPdmObjectScriptingCapability.h"
+#include "cafPdmUiDoubleSliderEditor.h"
+#include "cafPdmUiDoubleValueEditor.h"
+#include "cafPdmUiLineEditor.h"
+#include "cafPdmUiPushButtonEditor.h"
+#include "cafPdmUiTableViewEditor.h"
+#include "cafPdmUiTreeOrdering.h"
+#include "cvfGeometryTools.h"
+
+CAF_PDM_SOURCE_INIT( RimWellPathLateralGeometryDef, "WellPathLateralGeometryDef", "WellPathLateralGeometry" );
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+std::vector<RiaLineArcWellPathCalculator::WellTarget> createTargetsFromPoints( const std::vector<cvf::Vec3d>& points )
+{
+    CAF_ASSERT( points.size() >= 2u );
+
+    std::vector<RiaLineArcWellPathCalculator::WellTarget> targets;
+
+    for ( size_t i = 0; i < points.size(); ++i )
+    {
+        cvf::Vec3d tangent;
+        if ( i < points.size() - 1u )
+        {
+            tangent = points[i + 1] - points[i];
+        }
+        else if ( i > 0u )
+        {
+            tangent = points[i] - points[i - 1];
+        }
+        RiaOffshoreSphericalCoords sphericalCoords( tangent );
+
+        RiaLineArcWellPathCalculator::WellTarget target =
+            { points[i], true, sphericalCoords.azi(), sphericalCoords.inc(), 0.0, 0.0 };
+        targets.push_back( target );
+    }
+    return targets;
+}
+
+//--------------------------------------------------------------------------------------------------
+///
+//--------------------------------------------------------------------------------------------------
+RimWellPathLateralGeometryDef::RimWellPathLateralGeometryDef()
+    : changed( this )
+    , m_pickTargetsEventHandler( new RicCreateWellTargetsPickEventHandler( this ) )
+{
+    CAF_PDM_InitScriptableObjectWithNameAndComment( "Well Targets",
+                                                    ":/WellTargets.png",
+                                                    "",
+                                                    "",
+                                                    "WellPathLateralGeometry",
+                                                    "Class containing the geometry of a modeled Well Path Lateral" );
+
+    this->setUi3dEditorTypeName( RicWellPathGeometry3dEditor::uiEditorTypeName() );
+
+    CAF_PDM_InitScriptableField( &m_mdAtConnection, "MdAtConnection", 0.0, "MD at Well Path Connection", "", "", "" );
```

Comment:
I think we should try to find a better name for the connection location, consider `connectionMdOnParentWell`

Reply:
done